src/Utils.ml

```diff
@@ -58,3 +79,11 @@ let make_user_features feature_strings vars : (string * string) list =
                            let fdef = "(define-fun " ^ fname ^ sign ^ fs ^ ")"
                             in (fdef, fname))
    end
+
+let dedup_and_sort_keep_first ~compare list =
```

Comment:
As I explained in my ArrayComponents.ml comment, we basically need a `dedup_and_stable_sort` function. So change this function name accordingly and make it accept a `which_to_keep` argument.

Also, move it into the List module above (lines 10 -- 26).